From: Ben Hutchings <ben@decadent.org.uk>
Date: Mon, 27 Nov 2017 21:44:16 +0000
Subject: gpio: Avoid ABI change in 3.16.50
Forwarded: not-needed

Commit 14c8a620ba43 "gpio: drop retval check enforcing from
gpiochip_remove()" removed __must_check - which expands to
__attribute__((warn_unused_result)) - from the declaration of
gpiochip_remove().  This attribute doesn't affect the ABI, so add it
back for genksyms's eyes only.

---
--- a/include/linux/gpio/driver.h
+++ b/include/linux/gpio/driver.h
@@ -141,7 +141,11 @@ extern const char *gpiochip_is_requested
 
 /* add/remove chips */
 extern int gpiochip_add(struct gpio_chip *chip);
-extern int gpiochip_remove(struct gpio_chip *chip);
+extern int
+#ifdef __GENKSYMS__
+__must_check
+#endif
+gpiochip_remove(struct gpio_chip *chip);
 extern struct gpio_chip *gpiochip_find(void *data,
 			      int (*match)(struct gpio_chip *chip, void *data));
 
